### DISCUSSION TITLE
There is no way to update the changes in schema once tables are initialized (created) - DB migration feature

### DISCUSSION THREAD

**Discussion by nbaua** [jchristn/WatsonORM#10] [General]
Hi,

I have just came across this wonderful code and wanted to use for one of my SqlLite based personal project.
What I really liked about the API is it is super simple and the annotations are clear to create the schema/table at once.

However, I have noticed that once the tables are generated for the provided classes there is no way to update/upgrade them.
What I intend to achieve is a database versioning and/or migration capabilities.

If this things are working as of now, I am not sure if I have followed it correctly. Please suggest otherwise.
Thanks for this beautiful light-weight ORM, keep it rocking .. thanks.

**Comment by jchristn** [comment]
Hi @nbaua thanks for your kind words.  

It will be really hard to generate a table using WatsonORM.  The characteristics that define a table are metadata/annotations, not something that can be manipulated programmatically.  If you make table changes in the DB, you would need to update the metadata/annotations and recompile your code.

If you're looking to build out a database versioning/migration capability, may I suggest taking a look at the library that is the underpinning to WatsonORM - DatabaseWrapper.  It has more construct-level interaction with the underlying databases.

Cheers,
Joel

**Reply by jchristn** [reply]
I should also mention RestDb - which shows a model to expose programmatically the geometry of the database. https://github.com/jchristn/RestDb

Here is the link for DatabaseWrapper, sorry I forgot to include it. https://github.com/jchristn/DatabaseWrapper
